core/trino-spi/src/main/java/io/trino/spi/connector/RecordCursor.java

```diff
@@ -18,6 +18,9 @@
 
 import java.io.Closeable;
 
+import static io.trino.spi.connector.RecordCursor.AdvanceStatus.DATA_AVAILABLE;
```

Comment:
I looks we are tailoring `RecordCursor` interface for some specific use case.
Instead of modifying `RecordCursor` I suggest we just return pages using `ConnectorPageSource` (which supports yielding and blocking). Internally, we even do that (see `RecordPageSource`), so it should be easy to implement within connector